File: app/src/main/java/com/platdmit/simplecloudmanager/data/database/dao/ActionDao.java
package com.platdmit.simplecloudmanager.data.database.dao;

import com.platdmit.simplecloudmanager.data.database.entity.DbAction;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface ActionDao extends BaseDao<DbAction> {
    @Query("SELECT * FROM dbaction WHERE mResourceId = :serverId")
    List<DbAction> getActionForServer(long serverId);
}

File: app/src/main/java/com/platdmit/simplecloudmanager/data/database/dao/ActionDao.kt
package com.platdmit.simplecloudmanager.data.database.dao

import androidx.room.Dao
import androidx.room.Query
import com.platdmit.simplecloudmanager.data.database.entity.DbAction

@Dao
interface ActionDao : BaseDao<DbAction> {
    @Query("SELECT * FROM dbaction WHERE resourceId = :serverId")
    fun getActionForServer(serverId: Long): List<DbAction>?
}